### PR TITLE
Merge VoteFlow into codebase

### PR DESCRIPTION
**Pull request by Shiming94**
This pull request is to integrate VoteFlow into the OpenSceneFlow main branch. Readme and pretrained weights are also updated.

**Core Code**
- `src/models/voteflow.py`
- `src/basic/voteflow_plugin`
- `config/model/voteflow.yaml`

### CODE REVIEW

**Comment by Kin-Zhang** [comment]
It's not working in my environment somehow:
```
ch/include/ATen/CPUGeneratorImpl.h:3,\n                 from /home/kin/mambaforge/envs/opensf/lib/python3.8/site-packages/torch/include/ATen/Context.h:3,\n                 from /home/kin/mambaforge/envs/opensf/lib/python3.8/site-packages/torch/include/ATen/ATen.h:7,\n                 from /home/kin/data/code_online/OpenSceneFlow/src/models/basic/voteflow_plugin/hough_transformation/cpp_im2ht/ht_cuda.cu:2:\n/home/kin/mambaforge/envs/opensf/include/thrust/system/cuda/config.h:122:2: error: #error The version of CUB in your include path is not compatible with this release of Thrust. CUB is now included in the CUDA Toolkit, so you no longer need to use your own checkout of CUB. Define THRUST_IGNORE_CUB_VERSION_CHECK to ignore this.\n  122 | #error The version of CUB in your include path is not compatible with this release of Thrust. CUB is now included in the CUDA Toolkit, so you no longer need to use your own checkout of CUB. Define THRUST_IGNORE_CUB_VERSION_CHECK to ignore this.\n      |  ^~~~~\nninja: build stopped: subcommand failed.\n')
full_key: model.target
```

I will check again later.

**Comment by Kin-Zhang** [comment]
Hi @Shiming94 , could you update the README again about the training setup and hour for the model ckpt? Here: https://github.com/KTH-RPL/OpenSceneFlow/blob/feature/voteflow/README.md?plain=1#L147

File: src/models/basic/voteflow_plugin/hough_transformation/cpp_im2ht/ht_setup.sh
Comment (Kin-Zhang) [inline src/models/basic/voteflow_plugin/hough_transformation/cpp_im2ht/ht_setup.sh:1]: Do we use it? If not essential, maybe delete?

**Comment by Shiming94** [comment]
> It's not working in my environment somehow:
> 
> ```
> ch/include/ATen/CPUGeneratorImpl.h:3,
> from /home/kin/mambaforge/envs/opensf/lib/python3.8/site-packages/torch/include/ATen/Context.h:3,
> from /home/kin/mambaforge/envs/opensf/lib/python3.8/site-packages/torch/include/ATen/ATen.h:7,
> from /home/kin/data/code_online/OpenSceneFlow/src/models/basic/voteflow_plugin/hough_transformation/cpp_im2ht/ht_cuda.cu:2:\n/home/kin/mambaforge/envs/opensf/include/thrust/system/cuda/config.h:122:2: error: #error The version of CUB in your include path is not compatible with this release of Thrust. CUB is now included in the CUDA Toolkit, so you no longer need to use your own checkout of CUB. Define THRUST_IGNORE_CUB_VERSION_CHECK to ignore this.
> 122 | #error The version of CUB in your include path is not compatible with this release of Thrust. CUB is now included in the CUDA Toolkit, so you no longer need to use your own checkout of CUB. Define THRUST_IGNORE_CUB_VERSION_CHECK to ignore this.\n      |  ^~~~~\nninja: build stopped: subcommand failed.\n')
> full_key: model.target
> ```
> 
> I will check again later.
> 
> I see, it looks like we don't need to install these two packages:
> 
> ```shell
> conda install -c iopath iopath
> conda install -c bottler nvidiacub
> ```

Great that you made it run, when I setup the environment, I did need to install `liopath iopath bottler nvidiacub` to install pytorch-3d, maybe you already installed it when you installed the packages for other methods. I am okay with deleting it. 

> 
> It's working for me now: image
> 
> Validation results (Please note that this is evaluated **in the fixed validation set**, check more detail on fixing gt flow in #5 ):
> 
> The raw terminal result screenshot:

I checked #5, since the performance of all methods decreases a little bit, the current results are okay. I have double-checked the checkpoint before I made the PR, it is the best checkpoint of VoteFlow.

File: src/models/basic/voteflow_plugin/hough_transformation/cpp_im2ht/ht_setup.sh
Comment (Shiming94) [reply src/models/basic/voteflow_plugin/hough_transformation/cpp_im2ht/ht_setup.sh:1]: It is used to debug the CUDA file building, and it could be deleted.

**Review by Kin-Zhang — APPROVED** [review]
